pre_commit_hooks/string_fixer.py

```diff
@@ -32,7 +33,14 @@ def get_line_offsets_by_line_no(src):
 
 
 def fix_strings(filename):
-    contents = io.open(filename).read()
+    try:
+        contents = io.open(filename).read()
+    except Exception:
```

Comment:
I'd like to see if we can't just fix this hook -- there's others in this repository that have been converted to working with arbitrarily-encoded files by only handling files as binary.

Please write a test for such a fix when attempted :)  You may find the approach in #230 relevant.

This also shouldn't blanket catch `Exception` -- this has the potential to hide actual programming errors